packages/run-protocol/src/vaultFactory/liquidateIncrementally.js

```diff
@@ -0,0 +1,302 @@
+ * the amount of that collateral in the AMM pool and the desired price impact.
+ * It then gets 3 prices for the current tranche:
+ * - AMM quote - compute XYK locally based on the pool sizes
+ * - Reserve quote - based on a low price at which the Reserve will purchase
+ * - Oracle quote - provided by the oracle that triggered liquidation
+ * - Oracle limit - a limit that is DIVERGENCE_TOLERANCE below the
+ *   Oracle quote.
+ * Then sell based on the following:
+ * - If the AMM quote is *below* the oracle limit, skip this block
+ *   to wait for oracle and AMM prices to converge.
+ * - If the Reserve quote is higher than the AMM quote, sell to the
+ *   Reserve with a `want` of the Oracle limit
+ * - Otherwise, sell to the AMM with a `want` of the Oracle limit
+ *
+ * Selling uses the `oracleLimit as the `want` as the limit to allowed
+ * slippage, and provides the remaining `debt` as the `stopAfter` so
+ * that we sell no more than is needed to pay off the debt.
+ *
+ * TODO integrate the reserve, including the above Reserve strategies.
+ */
+
+/** @type {ContractStartFn} */
+const start = async zcf => {
+  const {
+    amm,
+    priceAuthority,
+    timerService,
+    debtBrand,
+    MaxImpactBP,
+    OracleTolerance,
+    AMMMaxSlippage,
+  } = /** @type {LiquidationContractTerms} */ zcf.getTerms();
+
+  trace('terms', zcf.getTerms());
+  const nextBlock = async () => E(timerService).delay(1n);
+
+  const SCALE = 1_000_000n;
+  const BASIS_POINTS = 10_000n * SCALE;
+  const BP2 = BASIS_POINTS * BASIS_POINTS;
+  const ONE = BASIS_POINTS;
+
+  const asFloat = (numerator, denominator) =>
+    Number(numerator) / Number(denominator);
+
+  /**
+   * Compute the tranche size whose sale on the AMM would have
+   * a price impact of MAX_IMPACT_BP.
+   * This doesn't use ratios so that it is usable for any brand
+   *
+   * @param {Amount} poolSize
+   * @param {bigint} maxImpactBP
+   * @param {bigint} feeBP
+   * @returns {Amount}
+   */
+  const maxTrancheWithFees = (poolSize, maxImpactBP, feeBP) => {
+    trace('maxTrancheWithFees', poolSize, maxImpactBP, feeBP);
+    const maxImpactScaled = maxImpactBP * SCALE;
+    const ammFeeScaled = feeBP * SCALE;
+    // sqrt(impact + 1) - 1
+    // Since we take SQRT but want to be BP, multiply in BP
+    const impactFactor =
+      BigInt(
+        Math.ceil(Math.sqrt(Number((maxImpactScaled + ONE) * BASIS_POINTS))),
+      ) - ONE;
+    const impactWithFee = impactFactor * (ONE - ammFeeScaled);
+    return AmountMath.make(
+      poolSize.brand,
+      NatMath.ceilDivide(NatMath.multiply(poolSize.value, impactWithFee), BP2),
+    );
+  };
+
+  function computeOracleLimit(oracleQuote, oracleTolerance) {
+    return ceilMultiplyBy(getAmountOut(oracleQuote), oneMinus(oracleTolerance));
+  }
+
+  const getAMMFeeBP = async () => {
+    const zoe = zcf.getZoeService();
+    const instance = await E(zoe).getInstance(E(amm).makeSwapInvitation());
+    const terms = await E(zoe).getTerms(instance);
```

Comment:
I don't think you need two awaits.
```suggestion
    const instanceP = E(zoe).getInstance(E(amm).makeSwapInvitation());
    const terms = await E(zoe).getTerms(instanceP);
```